assets/js/modules/analytics-4/datastore/conversion-reporting.js

```diff
@@ -226,6 +254,49 @@ export const selectors = {
 	 */
 	hasLostConversionReportingEvents:
 		hasConversionReportingEventsOfType( 'lostEvents' ),
+
+	/**
+	 * Checks if there are key metrics widgets connected with the detected events for the supplied purpose answer.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}  purpose      Value of saved site purpose from user input settings.
+	 * @param {boolean} useNewEvents Flag inclusion of detected new events, otherwise initial detected events will be used.
+	 * @return {boolean} TRUE if current site purpose will have any ACR key metrics widgets assigned to it, FALSE otherwise.
+	 */
+	haveConversionEventsForTailoredMetrics: createRegistrySelector(
+		( select ) => ( state, purpose, useNewEvents ) => {
```

Comment:
I removed `purpose` since currently it won't have usage, but we need `useNewEvents` as following issue that depends on this one will use that, and core infrastructure is laid down here